pkg/runtime/serializer/versioning/versioning.go

```diff
@@ -225,51 +137,95 @@ func (c *codec) Decode(data []byte, defaultGVK *unversioned.GroupVersionKind, in
 // Encode ensures the provided object is output in the appropriate group and version, invoking
 // conversion if necessary. Unversioned objects (according to the ObjectTyper) are output as is.
 func (c *codec) Encode(obj runtime.Object, w io.Writer) error {
-	if _, ok := obj.(*runtime.Unknown); ok {
+	switch t := obj.(type) {
+	case *runtime.Unknown:
+		if gv, ok := runtime.PreferredGroupVersion(c.encodeVersion); ok {
+			t.APIVersion = gv.String()
```

Comment:
Unknown is a specific case - for protobuf we're actually encoding the outer layer (the envelope with type info) directly into the output.  These are objects the scheme doesn't know about, so the scheme can't say anything about the type.